addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+    def _compute_is_donate(self):
+        for move in self:
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_carrier_type:
+                move.l10n_tw_edi_is_donate = False
+                move.l10n_tw_edi_love_code = False
+
+    @api.depends('l10n_tw_edi_is_print', 'l10n_tw_edi_is_donate')
+    def _compute_carrier_info(self):
+        for move in self:
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_is_donate:
+                move.l10n_tw_edi_carrier_type = False
+                move.l10n_tw_edi_carrier_number = False
+
+    @api.depends('line_ids.tax_ids')
+    def _compute_is_included_tax(self):
+        for move in self:
+            product_lines = move.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines())
+            move.l10n_tw_edi_ecpay_is_included_tax = "1" if any(tax.amount >= 5.0 for line in product_lines for tax in line.tax_ids) else "0"
+
+    # ----------------
+    # Business methods
+    # ----------------
+
+    # API methods
+
+    def _l10n_tw_edi_prepare_item_list(self, json_data):
+        item_list = []
+        for line in self.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines()):
+            taxable = line.tax_ids.filtered(lambda t: t.amount >= 5.0)
+            item_list.append(
+                {
+                    "ItemSeq": line.sequence,
+                    "ItemName": line.product_id.name[:100],
+                    "ItemCount": int(line.quantity),
+                    "ItemWord": line.product_uom_id.name[:6],
+                    "ItemPrice": int(line.price_total / line.quantity),
+                    "ItemTaxType": "1" if len(taxable.ids) >= 1 else "3",
+                    "ItemAmount": int(line.price_total),
+                }
+            )
+        json_data.update({"Items": item_list})
+
+    def _l10n_tw_edi_generate_invoice_json(self):
+        if not self.env.company.l10n_tw_edi_ecpay_merchant_id:
+            raise UserError(_("Please fill in the ECpay API information in the Setting!"))
+
+        # prepare json data
+        self.l10n_tw_edi_related_number = base64.urlsafe_b64encode(uuid.uuid4().bytes).decode()
+        json_data = {
+            "MerchantID": self.env.company.l10n_tw_edi_ecpay_merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+            "CustomerIdentifier": self.partner_id.vat or "",
+            "CustomerName": self.partner_id.name,
+            "CustomerAddr": self.partner_id.contact_address,
+            "CustomerEmail": self.partner_id.email or '',
+            "CustomerPhone": self.partner_id.phone or '',
+            "ClearanceMark": "",
+            "Print": str(int(self.l10n_tw_edi_is_print or self.partner_id.vat)),
+            "Donation": str(int(self.l10n_tw_edi_is_donate)),
+            "TaxType": "1" if self.l10n_tw_edi_ecpay_is_included_tax == "1" else "3",
```

Comment:
This seems a bit too simple. At least you should support "2" also (zero rated tax, so you have a tax on the invoice but its amount is 0%)